haystack/components/rankers/hugging_face_tei.py

```diff
@@ -0,0 +1,247 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from enum import Enum
+from typing import Any, Dict, List, Optional, Union
+from urllib.parse import urljoin
+
+from haystack import Document, component, default_from_dict, default_to_dict
+from haystack.utils import Secret, deserialize_secrets_inplace
+from haystack.utils.requests_utils import async_request_with_retry, request_with_retry
+
+
+class TruncationDirection(str, Enum):
+    """
+    Defines the direction to truncate text when input length exceeds the model's limit.
+
+    Attributes:
+        LEFT: Truncate text from the left side (start of text).
+        RIGHT: Truncate text from the right side (end of text).
+    """
+
+    LEFT = "Left"
+    RIGHT = "Right"
+
+
+@component
+class HuggingFaceTEIRanker:
+    """
+    Ranks a list of documents by their relevance to a given query using an external TEI reranking API.
+
+    This component sends the query and document texts to a configured TEI API endpoint
+    and retrieves a relevance-based ordering of the documents.
+
+    Usage example:
+    ```python
+    from haystack import Document
+    from haystack.components.rankers import HuggingFaceTEIRanker
+    from haystack.utils import Secret
+
+    reranker = HuggingFaceTEIRanker(
+        url="https://api.my-tei-service.com",
+        top_k=5,
+        timeout=30,
+        token=Secret.from_token("my_api_token")
+    )
+    docs = [Document(content="First doc"), Document(content="Second doc"), ...]
+    result = reranker.run(query="example query", documents=docs)
+    ranked_docs = result["documents"]
+    ```
+    """
+
+    def __init__(
+        self,
+        url: str,
```

Comment:
```suggestion
        self,
        *,
        url: str,
```

to avoid the errors we did in the past, let's force the usage of keyword arguments